pkg/kubectl/deployment.go

```diff
@@ -27,24 +27,42 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// DeploymentBasicGeneratorV1 supports stable generation of a deployment
-type DeploymentBasicGeneratorV1 struct {
+// BaseDeploymentGenerator: implement the common functionality of
```

Comment:
I don't feel strongly about it (that's why I said it's optional). I didn't see this pattern anywhere else in the codebase so it just caught my eye.